composer-base-components/map/map.tsx

```diff
@@ -0,0 +1,106 @@
+import { Map, Marker } from '@vis.gl/react-google-maps';
+import React, { useEffect, useState } from 'react';
+
+interface ComposerMapProps {
+  markers: { lat: number; lng: number; }[];
+  className: string;
+};
+
+type Coordinate = {
+  lat: number;
+  lng: number;
+};
+
+function ComposerMap({ markers, className }: ComposerMapProps) {
+  const [center, setCenter] = useState<Coordinate>(
+    {
+      lat: markers[0]?.lat ?? 36.8968908, // default center (antalya)
+      lng: markers[0]?.lng ?? 30.7133233,
+    }
+  );
+  const [restriction, setRestriction] = useState<google.maps.MapRestriction>(null);
```

Comment:
This feature prevents users from panning or scaling the map beyond certain boundaries. If there is such a feature, the user should be able to change it.